### ISSUE TITLE
[rogue] [vanish]

### ISSUE THREAD

**Issue by Jiaku** [dalaranwow/dalaran-wow#4922]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
Let's say you have hemorrhage and cheap shot on the same button. You press vanish and quickly pressing cheap shot, it will use hemorrhage instead (window where hemorrhage will be used instead of cheap shot is about 0.3s. I have tried same with macro where stances are invoved here is an exapmle (/cast [nomod,stance:1/3]Cheap Shot;[nomod,stance:0]Hemorrhage)(in macro you can see that if i'm in stealth or in shadow dance it won't use hemorrhage, it still uses hemorrhage if you press it too quickly.)
**Expected behaviour**: Tell us what should happen instead
You can use Cheap Shot right away, without any delay
**Steps to reproduce the problem**:

1. Stand next to dummy
2. Press vanish
3. immediately press hemorrhage/cheap shot button 

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Zarjofgt** [comment]
This is an issue with druid forms as well. Even when powershifting.

**Comment by Zarjofgt** [comment]
Could not reproduce this issue today, my guess is that it occurs during server delay or something.

**Comment by Jiaku** [comment]
It  has to an issue everywhere where is stance changing is invoved.

**Comment by Jiaku** [comment]
Any news?

**Comment by Aridam** [comment]
Confirming this with Stances at the very least -- as a Warrior, if I time my Stance switches (with and without macros) as close as I can to the GCD triggered by the Stance switching, at times I get "Ability is not ready yet" and it fails -- despite not doing it too soon, and the icon actually lights up -- or, it does go through, but with a delay about as described originally.

**Comment by Aridam** [comment]
Anything on this? I personally still get the delay/error with Stances/macros on Live.

**Comment by Aridam** [comment]
Ah. Makes sense.

**Comment by Jiaku** [comment]
I can't tell for warriors but as a rogue i can say it is game breaking, let's say you have several dots on you ticking and you need quickly vanish and perform a cheapshot but intead you have to wait a bit to do a cheapshot and from expience i can say that you will get out of stealth. I may be wrong on this but as i understand vanish is a self applied spell same goes to stance switching refering to #3680 so why there is a delay?

**Comment by Aridam** [comment]
It affects practically any and all Stance-based action -- be it Warrior, Druid, Rogue, et cetera. So yes it applies to Vanish and as a Stealth Class such as a Rogue relying on an instant re-Stealth that is Vanish, it _can_ be in fact gamebreaking. And as far as I also personally remember, such effects are self-applied and thus should be instant.

Macros-wise I don't use that type of command with the exclamation but I do use macros for almost everything I do and that includes even Stances -- switching Stances manually _and_ with macros provides the same effect, which is delay, and trying to queue actions does the same thing.

I'm certain what Teridan said above us is exactly the reason as to why this pretty much happens -- I had forgotten about that and that makes absolute sense as to how it worked on Retail, whereas here, that's not implemented; that must be why this is failing -- there's no queue system to train more than one Spell over a short period of time here.

**Comment by Jiaku** [comment]
Any news?

**Comment by Jiaku** [comment]
I have found one "workaround" via macro with /castsequence  reset=0,5 it forces to double tap on button to prevent getting out of vanish with hemorrhage instead of cheapshot. But it means you have to sit in vanish extra 0.5 seconds where your opponent could react or dot you get you out. 
As @TeridanDW suggested there may be queue system but that doesn't make it any faster you still have to wait extra 0.5 sedonds before you can perform cheapshot after vanish.

**Comment by Jiaku** [comment]
Any news?

**Comment by Jiaku** [comment]
Any news, thoughts, ideas?

**Comment by Jiaku** [comment]
I have an idea, if you can get rid of spell delay system that imlanted on the server the problem with all stances will be gone, instead of making new system that can screw things up.

**Comment by robinsch** [comment]
No idea how to confirm because it will always cheap shot the target when I'm testing this

**Comment by Jiaku** [comment]
Exactly, since it is local server there would be no delay... That's why you dont have any issues. Thats not even close to proper testing. Server where we play has server delay, your local server doesn't have it and even if it does you wont have any issues since it is inside your mashine...

**Comment by Jiaku** [comment]
>  On retail you had a delay between 1-400ms because all non self casts where handled in batches. 

> 

This is your comment regarding spell delay. Stance switching should be self apllied, there should be no delay but there is and this is an issue. Plus there TONs other spell that are self applied and yet they have delay.

**Comment by Jiaku** [comment]
WTB fix

**Comment by Jiaku** [comment]
5 months, no fix :(

**Comment by Jiaku** [comment]
Any sign of progess?

**Comment by Aridam** [comment]
Reconfirming this as still bugged the way it's been after a while -- Stances and other self-applied Auras suffer a delay, while any actions following the Global Cooldown sometimes lag-behind and cause a ~0.1s stutter inbetween Ability usage if done back-to-back (either via button-mashing or precisely-timed keypresses); refer to the referenced issue for more information on this which relates to this report.

**Comment by Aridam** [comment]
Any progress on the issue at hand? It'd be greatly-appreciated because this corresponds to pretty-much a fundamental part of the game.

Having no Spell Queue makes using Abilities back-to-back abysmal because of the random GCD error, and trying to use Abilities like Overpower just as the Proc window for it fades-away, I get the error "This ability requires combo points" and fail to use it because the server doesn't handle its request when I do it in the nick of time, due to Spell Delay affecting it first (as a Target-based Ability), but no Spell Queue to handle it on-time.

Stances and other Self-Applied Auras being affected by Spell Delay when they shouldn't, is also a bother-and-a-half, because it slows-down response timers with those as well due to them being delayed in a way they shouldn't be.

**Comment by Jiaku** [comment]
bummer that such an important aspect of the game doesn't get fix, as a PvP player this kinda bug scares you away from the server. It is not an enjoyable expirience when you know you "had it/pressed it" but you didn't get spell off that should be instant.

**Comment by Aridam** [comment]
The thing is both are rather complex -- least I assume so, logically-speaking -- and pretty-in-depth as well, so it shouldn't come as that much of a surprize; example is yes Spell Delay _is_ implemented here, but the catch is, it's not 100% correct -- just needs a fix on Self-Applied Auras to not get affected by Spell Delay and done. Spell Queue on the other hand, is a whole different story on its own, so as much time and resourcing as it may need in order for it to be implemented and _properly_ at that, both it _and_ fixing Spell Delay should get.

The implications of this whole ordeal go way-beyond PvP alone -- high-end PvE also suffers from this due to not being able to fully-optimally execute your rotations and react as quickly as you think you are allowed to; if it's not Spell Delay on Self-Applied Auras, then it's the GCD error if you're using Abilities too quickly/back-to-back, and if it's not that then it's backwards, and sometimes even both.

This should probably be at the top of the priority list for being such a core aspect of the game as a whole, but as long as it needs for these issues to be _proper-fixed_, then so be it.

**Comment by Aridam** [comment]
Anything to update with on the matter?

**Comment by Jiaku** [comment]
MIght as well close the topic, since it is not getting fix

**Comment by Aridam** [comment]
Bump.

Looks like this issue hasn't been touched upon, and the report is ever-so-nearing two-years-old -- quite a bit, for an issue this big.

https://www.youtube.com/watch?v=MyU2g5DkJPE

Even trying to use my Ranged Weapon sometimes causes this.

https://www.youtube.com/watch?v=7_JEwinGhyQ

Even my goddamned Stances.

https://www.youtube.com/watch?v=1MvXZ3RVT8c

This is a video showing me trying to recreate an ordinary instance of this bug -- trying to cycle Abilities as you would in any rotation. Near the end you can see this smooth chain get _absolutely ruined_ when my happy-go-lucky Rend spam gets stopped _and_ some random mini-GCD error disallowing me to use anything else in that fraction of a second just pops in and goes "HEY, HOW'YA DOIN', WANNA GET YOUR DAY RUINED?"
I don't think it listened to me saying "NO."
Side-note: I'm actually lucky to have gone that far in my spamming before this bug occurred -- normally it happens pretty soon and often, so this doesn't do it as much justice until you try it for yourselves enough times.

Also, this can happen with Abilities that don't share or trigger a GCD -- like Sweeping Strikes.
Sometimes, I would try to use Sweeping Strikes the _moment_ its Cooldown finishes. And then it just _decides_ that no, it's not actually ready. When that happens it's pretty much like the first video -- showing the error when using my Ranged Weapon: it lights up and "pretends" to use it, but then stops, doesn't actually use it, returns an error that it's "not ready", Combat Log reports that I "haven't recovered", so suddenly I have a game telling me I'm still tired and should take a nap.
